assets/js/bracket.js

```diff
@@ -0,0 +1,152 @@
+import "../css/bracket.scss";
+import $ from "jquery";
+
+const SELECTOR = "#brackets-viewer-container";
+const TAB_CONFIG = [
+  ["list-view-tab", "list-view"],
+  ["bracket-view-tab", "bracket-view"]
+];
+
+let matchMetadata = {};
+
+const reshapeParticipants = (participants = [], useTeamNames) =>
+  participants.map(entry => {
+    const name = entry.team_name;
+    const debaters = entry.debaters_names || "";
+    return { ...entry, name: useTeamNames ? name : debaters.trim() || name };
+  });
+
+const renderTeamBlock = (label, info) => {
+  const name = info && info.display ? info.display : "TBD";
+  const members =
+    info && info.debaters
+      ? `<div class="team-members text-muted small mt-1">${info.debaters}</div>`
+      : "";
+  return (
+    `<div class="match-team mb-3"><strong>${label}</strong>` +
+    `<span class="team-name d-block font-weight-bold">${name}</span>` +
+    `${members}</div>`
+  );
+};
+
+const renderBracket = () => {
+  const viewer = window.bracketsViewer;
+  const $container = $(SELECTOR);
+  const payload = window.bracketViewerData;
+  if (!viewer || !viewer.render || !$container.length || !payload) return;
+
+  const useTeamNames = $("body").hasClass("show-team-names");
+  const data = {
+    ...payload,
+    participants: reshapeParticipants(payload.participants, useTeamNames)
+  };
+  viewer.render(data, { selector: SELECTOR, clear: true });
+  matchMetadata = payload.match_metadata || {};
+};
+
+window.renderBracket = renderBracket;
+
+const activateTab = (targetPane, { render = false } = {}) => {
+  TAB_CONFIG.forEach(([tab, pane]) => {
+    const $tab = $(`#${tab}`);
+    const $pane = $(`#${pane}`);
+    const active = pane === targetPane;
+
+    $tab.toggleClass("active", active).attr("aria-selected", active);
+    $pane.toggleClass("active", active);
+  });
+
+  const showBracket = targetPane === "bracket-view";
+  $("body").toggleClass("bracket-view-active", showBracket);
+  if (showBracket && render) renderBracket();
+};
+
+const openModal = metadata => {
+  const $modal = $("#match-metadata-modal");
+  const $backdrop = $("#match-modal-backdrop");
+  const $content = $("#modal-content");
+  if (!$modal.length || !$backdrop.length || !$content.length) return;
+
+  const judges =
+    metadata.judges && metadata.judges.length
+      ? metadata.judges
+          .map(judge => {
+            const name = judge.is_chair
+              ? `<strong>${judge.name}</strong>`
```

Comment:
This likely isn't a threat we're at risk for, but be warned that this makes us vulnerable to XSS/HTML injenction. Probably work sanitizing